packages/angular_devkit/build_webpack/test/browser/service-worker_spec_large.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { join, normalize, virtualFs } from '@angular-devkit/core';
+import { tap } from 'rxjs/operators';
+import { browserTargetSpec, host, runTargetSpec } from '../utils';
+
+
+fdescribe('Browser Builder', () => {
+  const outputPath = normalize('dist');
+
+  beforeEach(done => host.initialize().subscribe(undefined, done.fail, done));
+  afterEach(done => host.restore().subscribe(undefined, done.fail, done));
+
+  it('errors if no ngsw-config.json is present', (done) => {
+    const overrides = { serviceWorker: true };
+
+    try {
+      runTargetSpec(host, browserTargetSpec, overrides).subscribe(undefined, done, done.fail);
+    } catch (e) {
+      // Because this is synchronous.
+      done();
+    }
+  }, 30000);
+
+  it('works with service worker', (done) => {
+    host.writeMultipleFiles({
+      'src/ngsw-config.json': JSON.stringify({
+          index: '/index.html',
+          assetGroups: [{
+            name: 'app',
+            installMode: 'prefetch',
+            resources: {
+              files: [
+                '/favicon.ico',
+                '/index.html',
+              ],
+              versionedFiles: [
+                '/*.bundle.css',
+                '/*.bundle.js',
+                '/*.chunk.js',
+              ],
+            },
+          }, {
+            name: 'assets',
+            installMode: 'lazy',
+            updateMode: 'prefetch',
+            resources: {
+              files: [
+                '/assets/**',
+              ],
+            },
+          }],
+        }),
+    });
+
+    const overrides = { serviceWorker: true };
+
+    runTargetSpec(host, browserTargetSpec, overrides).pipe(
+      tap((buildEvent) => {
+        expect(buildEvent.success).toBe(true);
```

Comment:
It would be good if we also added the other checks as in the origin e2e test:

- check the file `ngsw.json` is generated correctly as in the original test? https://github.com/angular/angular-cli/blob/d1b22d0e680f254b8395ad48ed1c8f29c9b4d46f/tests/e2e/tests/build/service-worker.ts#L39

- `base-href` should be taken into account https://github.com/angular/angular-cli/blob/d1b22d0e680f254b8395ad48ed1c8f29c9b4d46f/tests/e2e/tests/build/service-worker.ts#L40-L42

- the safety-worker check https://github.com/angular/angular-cli/blob/d1b22d0e680f254b8395ad48ed1c8f29c9b4d46f/tests/e2e/tests/build/service-worker.ts#L45-L48

Reply:
I'm not done with this PR.